src/Tribe/Repository.php

```diff
@@ -3167,20 +3169,38 @@ protected function build_query_internally() {
 		 */
 		$query_args = apply_filters( "tribe_repository_{$this->filter_name}_query_args", $query_args, $query, $this );
 
-		if ( isset( $query_args['offset'] ) ) {
-			$offset = absint( $query_args['offset'] );
+		/**
+		 * Provides a last-ditch effort to override the filtered offset.
+		 *
+		 * This should only be used if doing creating pagination for performance purposes.
+		 *
+		 * @since 5.0.0
+		 *
+		 * @param null|int Offset parameter setting.
+		 */
+		$filtered_offset = apply_filters( 'tribe_repository_query_arg_offset_override', null, $query_args );
+
+		if ( $filtered_offset || isset( $query_args['offset'] ) ) {
 			$per_page = (int) Tribe__Utils__Array::get( $query_args, 'posts_per_page', get_option( 'posts_per_page' ) );
-			$page = (int) Tribe__Utils__Array::get( $query_args, 'paged', 1 );
 
-			$real_offset = $per_page === - 1 ? $offset : ( $per_page * ( $page - 1 ) ) + $offset;
-			$query_args['offset'] = $real_offset;
-			$query_args['posts_per_page'] = $per_page === - 1 ? 99999999999 : $per_page;
+			if ( $filtered_offset ) {
+				$query_args['offset'] = $filtered_offset;
+			} elseif ( isset( $query_args['offset'] ) ) {
+				$offset   = absint( $query_args['offset'] );
+				$per_page = (int) Tribe__Utils__Array::get( $query_args, 'posts_per_page', get_option( 'posts_per_page' ) );
```

Comment:
Should the posts_per_page option be called every time even if posts_per_page is set?

Reply:
This line should actually be removed as it is present on line 3185! (I'll remove it!)

Regarding your actual question: that's a good one and perhaps something to dig into–though not as part of this PR. I'd like to leave as much unchanged as possible to avoid potential bugs that result from non-critical changes.